### DISCUSSION TITLE
`cache.resolve` seems to invalidate query

### DISCUSSION THREAD

**Discussion by sheldonbaker** [urql-graphql/urql#2013] [Q&A] [answered]
I'm having an odd issue with `cache.resolve`.

I have the name of a field on my `Query` root (`firstFieldKey` in the example below), and I'm trying to put a newly inserted object at the head of the list after a mutation. That all seems to work, except that the query seems to be invalidated and hits the network when I navigate to a page that uses that query. I'm expecting the query to not reload, and to just see my cached results.

Most confusingly, simply modifying the data from `cache.resolve` (not even putting the data back into the cache with `cache.link`) seems to be the culprit. The object I am inserting seems to have data for all the fields that the query is asking for.

Am I misunderstanding something, or is there an easy way I can see how/why/where that field seems to be invalidated?

```
updates: {
  Mutation: {
    favoriteInsert({ favoriteInsert }, { object }, cache) {
      // ommitted ...

      let data = cache.resolve('Query', firstFieldKey)

      // data.unshift(favoriteInsert) // uncommenting THIS line (not the following line) seems to invalidate the `firstFieldKey` query (navigating to a page where that query was used before sees the whole query hit the network again)
      
      // this works as expected, except for the aforementioned problem
      // cache.link('Query', firstFieldKey, data)
      // console.log(cache.resolve('Query', firstFieldKey)) // works as expected - I see the list of items with the new item at the top

      // ....

**Comment by JoviDeCroock** [comment, marked as answer]
Most likely it'd be a cache-miss due to some missing field, you can place the `debugExchange` in front of the cacheExchange to get an idea of how the operations are flowing. Cache.resolve itself has no destructive events normally 😅 sorry if this isn't of much help, can just throw the general idea as we'd need to see the data - query - mutation and full updater to paint a good picture.

**Comment by sheldonbaker** [comment]
Okay, I'll check that out.

Do you know why calling unshift on the data returned by resolve would cause it? I was under the impression that that data would just be a mere copy of the cache's data.

Thanks!

**Reply by JoviDeCroock** [reply]
Have you tried doing an immutable return instead?

**Comment by sheldonbaker** [comment]
Well, I stepped away from the computer for a few hours and, when I came back, all was working as expected, hah.

Not sure what it was, ‘cause I can’t replicate it now. Thanks!
